src/gluonts/ext/r_forecast/_predictor.py

```diff
@@ -318,14 +324,37 @@ def predict(
         """
         self._warning_message()
 
+        len_ts = len(dataset[0]['target'])
+        freq = dataset[0]['start'].freq
+        period = get_seasonality(freq)
+
+        fourier_ratio = len_ts / period
+        fourier = False
+        if ((period > fourier_frequency_low_periods and fourier_ratio > fourier_ratio_threshold_low_periods) or
+                (period >= fourier_frequency_high_periods and fourier_ratio > fourier_ratio_threshold_high_periods)):
+            fourier = True
+            K = min(fourier_order, np.floor(period / 2))
+            xreg_full = []
+            for k in range(1, K + 1):
+                xreg_full.append(np.sin(2 * np.pi * k / period * np.arange(1, len_ts + self.prediction_length + 1)))
+                xreg_full.append(np.cos(2 * np.pi * k / period * np.arange(1, len_ts + self.prediction_length + 1)))
+            xreg_full = np.transpose(np.array(xreg_full, dtype=np.float32))
+            xreg = xreg_full[:-self.prediction_length, :]
+            xreg_future = xreg_full[-self.prediction_length:, :]
+
+        if self.method_name == 'arima' and fourier:
+            self.params["seasonal"] = False
+            self.params["xreg"] = xreg
+            self.params["xreg_future"] = xreg_future
```

Comment:
I think this should be a separate function, so that we can test it and benchmark it. Also, features should be computed for each dataset entry independently, since they may have different length/frequency